FirebaseInstallations/Source/Library/FIRInstallationsItem.m

```diff
@@ -52,4 +52,30 @@ + (NSString *)identifierWithAppID:(NSString *)appID appName:(NSString *)appName
   return [appID stringByAppendingString:appName];
 }
 
++ (NSString *)generateFID {
+  NSUUID *uuid = [NSUUID UUID];
+  uuid_t uuidBytes;
+  [uuid getUUIDBytes:uuidBytes];
+
+  NSData *uuidData = [NSData dataWithBytes:uuidBytes length:16];
+
+  uint8_t prefix = 0b01110000;
+  NSMutableData *fidData = [NSMutableData dataWithBytes:&prefix length:1];
+
+  [fidData appendData:uuidData];
+  NSString *fidString = [fidData base64EncodedStringWithOptions:0];
+
+  // TODO: Consider implementation which does not modify UUID.
+
+  // A valid FID has exactly 22 base64 characters, which is 132 bits, or 16.5 bytes.
+  // Our generated ID has 16 bytes UUID + 1 byte prefix which after encoding with base64 will become
+  // 23 characters plus 1 character for "=" padding.
+
+  // Remove the 23rd character that was added because of the extra 4 bits at the
+  // end of our 17 byte data. It should be pretty safe to do because UUID ends with the random part,
+  // so we will not affect probability of the collisions much.
+  // Also remove the '=' padding.
+  return [fidString substringWithRange:NSMakeRange(0, 22)];
```

Comment:
Make sure to generate a test FID and check with Rayo to double check the format.